presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java

```diff
@@ -16,8 +16,18 @@
 import com.facebook.presto.spi.block.Block;
 
 public interface ConnectorPageSink
+        extends AutoCloseable
 {
     void appendPage(Page page, Block sampleWeightBlock);
 
     String commit();
+
+    /**
+     * This method is guaranteed to be called after commit or on failure.
```

Comment:
We need to document what happens if an exception is thrown here. Does it abort the query (i.e. cancel the write)? @dain?